zap/src/output/luau/client.rs

```diff
@@ -70,6 +70,52 @@ impl<'src> ClientOutput<'src> {
 		}
 	}
 
+	fn push_event_loop(&mut self) {
+		self.push_line("");
+
+		let delta_time = if self.config.manual_event_loop {
+			self.config.casing.with("DeltaTime", "deltaTime", "delta_time")
+		} else {
+			"dt"
+		};
+
+		if self.config.manual_event_loop {
+			let send_events = self.config.casing.with("SendEvents", "sendEvents", "send_events");
+
+			self.push_line(&format!("local function {send_events}({delta_time}: number)"));
+		} else {
+			self.push_line(&format!("RunService.Heartbeat:Connect(function({delta_time})"));
+		}
+
+		self.push_line(&format!(
+			r#"	time += {delta_time}
+
+	if time >= 1 / 61 then
+		time -= 1 / 61
+
+		if outgoing_used ~= 0 then
+			local buff = buffer.create(outgoing_used)
+			buffer.copy(buff, 0, outgoing_buff, 0, outgoing_used)
+
+			reliable:FireServer(buff, outgoing_inst)
+
+			outgoing_buff = buffer.create(64)
+			outgoing_used = 0
+			outgoing_size = 64
+			table.clear(outgoing_inst)
+		end
+	end"#,
```

Comment:
This should be split up into mutiple lines w. `self.push_line` and `self.indent` etc. Use the rest of the file as an example.